tests/Unit/Parallel/Actions/Test_Goto.cpp

```diff
@@ -49,4 +92,29 @@ SPECTRE_TEST_CASE("Unit.Parallel.GotoAction", "[Unit][Parallel][Actions]") {
   runner.force_next_action_to_be<component, Actions::Goto<Label2>>(0);
   runner.next_action<component>(0);
   CHECK(runner.get_next_action_index<component>(0) == 1);
+
+  runner.set_phase(Metavariables::Phase::TestRepeatUntil);
+  while (not ActionTesting::get_terminate<component>(runner, 0)) {
+    runner.next_action<component>(0);
+  }
+  CHECK(ActionTesting::get_databox_tag<component, HasConverged>(runner, 0));
+  CHECK(ActionTesting::get_databox_tag<component, Counter>(runner, 0) == 2);
+
+  // Test zero iterations of the `RepeatUntil` loop
+  runner.set_phase(Metavariables::Phase::TestRepeatUntil);
+  runner.template algorithms<component>().at(0).set_terminate(false);
+  size_t number_of_actions_called = 0;
+  while (not ActionTesting::get_terminate<component>(runner, 0)) {
+    runner.next_action<component>(0);
+    number_of_actions_called++;
+  }
+  // Make sure `RepeatUntil` and `Parallel::Actions::TerminatePhase` were
+  // called. This is an inequality because `RepeatUntil` should have at least
+  // one action call, but could have more that the public interface does not
+  // care about.
+  CHECK(number_of_actions_called >= 2);
```

Comment:
This should be a strong equality if you actually want this check to be meaningful... As it stands it doesn't address what @kidder was worried about.

Reply:
This check makes sure that the action is called. That's what I understood @kidder asked. The check below (testing that the counter has not incremented any further) makes sure the body of the `RepeatUntil` loop is not executed. I think this actually tests everything, don't you agree?